supernode/daemon/mgr/ha/etcd_tool.go

```diff
@@ -0,0 +1,152 @@
+package ha
+
+import (
+	"context"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/supernode/config"
+
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/etcd/clientv3"
+)
+
+//EtcdMgr is the struct to manager etcd.
+type EtcdMgr struct {
+	config            clientv3.Config
+	client            *clientv3.Client
+	leaseTTL          int64
+	leaseKeepAliveRsp <-chan *clientv3.LeaseKeepAliveResponse
+	hostIP            string
+	leaseResp         *clientv3.LeaseGrantResponse
+}
+
+const (
+	//ActiveSupernodeOFF means there is no active supernode.
+	ActiveSupernodeOFF = ""
+	//ActiveSupernodeChange means active supernode change to standby supernode because of unhealthy.
+	ActiveSupernodeChange = 0
+	//ActiveSupernodeKeep means active supernode is health.
+	ActiveSupernodeKeep = 1
+	//etcdTimeOut is the etcd client's timeout second
+	etcdTimeOut = 10 * time.Second
+	//activeKeyPreFIx is the keyPrefix of active supernode
+	activeKeyPreFIx = "/supernode/active/"
+	//standbyKeyPreFix is the keyPrefix of active supernode
+	//standbyKeyPreFix = "/supernode/standby/"
+)
+
+//NewEtcdMgr produce a etcdmgr object.
+func NewEtcdMgr(cfg *config.Config) (*EtcdMgr, error) {
+	config := clientv3.Config{
+		Endpoints:   cfg.HAConfig,
+		DialTimeout: etcdTimeOut,
+	}
+	// build connection to etcd.
+	client, err := clientv3.New(config)
+	return &EtcdMgr{
+		hostIP: cfg.AdvertiseIP,
+		config: config,
+		client: client,
+	}, err
+}
+
+//WatchActiveChange is the progress to watch the etcd,if the value of key /lock/active changes,supernode will be notified.
```

Comment:
`/lock/active` or `/supernode/active/`? BTW, I think maybe `activeKeyPreFIx` is better.